libsweep/src/sweep.cc

```diff
@@ -288,111 +235,72 @@ void sweep_device_accumulate_scans(sweep_device_s device) {
       received = 1;
     }
   }
+} catch (const std::exception& e) {
+  // worker thread is dead at this point
```

Comment:
`accumulate_scans` does not have an out error argument:

https://github.com/scanse/sweep-sdk/blob/cd9b06e36e80140af373d85ca5a1abb47ea0a2b7/libsweep/include/sweep/sweep.h#L65-L66

So it's not possible to propagate it to the user. Maybe this was an oversight?